### PR TITLE
Protocol analysis/renames

### PR DESCRIPTION
**Pull request by Jaguar0625**
should address most remaining proposals that don't break binary protocol

### CODE REVIEW

File: schemas/mosaic/mosaic_supply_change.cats [hunk 1]
```diff
@@ -6,8 +6,8 @@ struct MosaicSupplyChangeTransactionBody
 	# id of the affected mosaic
 	mosaicId = UnresolvedMosaicId
 
-	# supply change direction
-	direction = MosaicSupplyChangeDirection
+	# supply change action
+	direction = MosaicSupplyChangeAction
```
Comment (Jaguar0625) [inline schemas/mosaic/mosaic_supply_change.cats]: not sure if direction should be renamed as well

File: schemas/multisig/multisig_account.cats [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import "multisig/multisig_types.cats"
 import "transaction.cats"
 
-# binary layout for a modify multisig account transaction
-struct ModifyMultisigAccountTransactionBody
+# binary layout for a multisig account transaction
+struct MultisigAccountTransactionBody
```
Comment (Jaguar0625) [inline schemas/multisig/multisig_account.cats]: unsure about name, but for consistency all tx types should end in Transaction

File: schemas/namespace/namespace.cats [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import "namespace/namespace_types.cats"
+import "transaction.cats"
+
+# binary layout for a namespace transaction
+struct NamespaceTransactionBody
+	# type of the registered namespace
+	disposition = NamespaceType
```
Comment (Jaguar0625) [inline schemas/namespace/namespace.cats]: not sure what to call this? `type` conflicts with `transaction.type`.

File: schemas/restriction_account/restriction_account_types.cats [hunk 4]
```diff
@@ -35,14 +35,14 @@ struct AccountAddressRestrictionModification
 	inline AccountRestrictionModification
 
 	# address restriction value
-	value = Address
+	value = UnresolvedAddress
```
Comment (Jaguar0625) [inline schemas/restriction_account/restriction_account_types.cats]: this is actually handled correctly by server already

File: schemas/multisig/multisig_types.cats [hunk 5]
```diff
@@ -11,7 +11,7 @@ enum CosignatoryModificationType : uint8
 # binary layout for a cosignatory modification
 struct CosignatoryModification
 	# modification type
-	modificationType = CosignatoryModificationType
+	modificationType = CosignatoryModificationAction
```
Comment (Jaguar0625) [inline schemas/multisig/multisig_types.cats]: should `modificationType` be renamed as well?

File: schemas/namespace/namespace.cats (hunk 3 above)
Comment (rg911) [reply schemas/namespace/namespace.cats]: `NamespaceLevel` ??
But also think `NamespaceType` should be fine

File: schemas/metadata/account_metadata.cats [hunk 6]
```diff
@@ -15,6 +15,7 @@ struct AccountMetadataTransactionBody
 	valueSize = uint16
 
 	# value data
+	# \note when updating, this value is XORed with current value
```
Comment (rg911) [inline schemas/metadata/account_metadata.cats]: what does? `XORed` stand for?
Should this comment line be displayed on generated file ?

File: schemas/metadata/namespace_metadata.cats [hunk 7]
```diff
@@ -19,6 +19,7 @@ struct NamespaceMetadataTransactionBody
 	valueSize = uint16
 
 	# value data
+	# \note when updating, this value is XORed with current value
```
Comment (rg911) [inline schemas/metadata/namespace_metadata.cats]: same above

File: schemas/metadata/mosaic_metadata.cats [hunk 8]
```diff
@@ -18,6 +18,7 @@ struct MosaicMetadataTransactionBody
 	valueSize = uint16
 
 	# value data
+	# \note when updating, this value is XORed with current value
```
Comment (rg911) [inline schemas/metadata/mosaic_metadata.cats]: same above

File: schemas/mosaic/mosaic_supply_change.cats (hunk 1 above)
Comment (rg911) [reply schemas/mosaic/mosaic_supply_change.cats]: My personal preference is `Action`
`Action` -> `Increase / Decrease'
`Direction` -> `Up / Down`  ???

File: generators/cpp_builder/hints/includes.yaml [hunk 9]
```diff
@@ -3,5 +3,5 @@
 MosaicDefinitionTransaction:
   - plugins/txes/mosaic/src/model/MosaicIdGenerator.h
 
-RegisterNamespaceTransaction:
+NamespaceTransaction:
```
Comment (dgarcia360) [inline generators/cpp_builder/hints/includes.yaml]: NamespaceRegistrationTransaction

File: generators/cpp_builder/hints/namespaces.yaml [hunk 10]
```diff
@@ -19,7 +19,7 @@ MosaicDefinitionTransaction:
 MosaicSupplyChangeTransaction:
   MosaicSupplyChangeDirection: model
 
-ModifyMultisigAccountTransaction:
+MultisigAccountTransaction:
```
Comment (dgarcia360) [inline generators/cpp_builder/hints/namespaces.yaml]: MultisigAccountModificationTransaction

File: generators/cpp_builder/hints/namespaces.yaml [hunk 11]
```diff
@@ -33,7 +33,7 @@ TransferTransaction:
   Mosaic: model
   UnresolvedMosaic: model
 
-RegisterNamespaceTransaction:
+NamespaceTransaction:
```
Comment (dgarcia360) [inline generators/cpp_builder/hints/namespaces.yaml]: NamespaceRegistrationTransaction*

File: schemas/namespace/namespace.cats (hunk 3 above)
Comment (dgarcia360) [reply schemas/namespace/namespace.cats]: Taking as an example "AccountAddressRestrictionTransactionBody", this has a property ``restrictionType``. 

We could achieve something similar by naming:

* disposition --> registrationType
* NamespaceType -> NamespaceRegistrationType

File: schemas/multisig/multisig_account.cats (hunk 2 above)
Comment (BloodyRookie) [reply schemas/multisig/multisig_account.cats]: yea, MultisigAccountModificationTransactionBody sounds good to me

File: schemas/namespace/namespace.cats (hunk 3 above)
Comment (BloodyRookie) [reply schemas/namespace/namespace.cats]: NamespaceRegistrationTransactionBody imo

File: schemas/account_link/account_link.cats [hunk 12]
```diff
@@ -3,8 +3,8 @@ import "transaction.cats"
 
 # binary layout for an account link transaction
 struct AccountLinkTransactionBody
-	# remote account key
-	remoteAccountKey = Key
+	# remote account public key
+	remoteAccountPublicKey = Key
```
Comment (gimer) [inline schemas/account_link/account_link.cats]: this will break cpp builders (or rather transaction generation)

Comment (Jaguar0625) [reply schemas/account_link/account_link.cats]: yes, cpp code will need to be updated once approved

File: schemas/mosaic/mosaic_supply_change.cats [hunk 13]
```diff
@@ -3,11 +3,11 @@ import "transaction.cats"
...
 # binary layout for a mosaic supply change transaction
 struct MosaicSupplyChangeTransactionBody
-	# id of the affected mosaic
+	# affected mosaic id
 	mosaicId = UnresolvedMosaicId
 
-	# supply change direction
-	direction = MosaicSupplyChangeDirection
+	# supply change action
+	actuon = MosaicSupplyChangeAction
```
Comment (gimer) [inline schemas/mosaic/mosaic_supply_change.cats]: typo

**Comment by Jaguar0625** [comment]
Thanks for reviewing!

**Pull request closed without merging**